exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -56,7 +58,7 @@ def initialize(endpoint: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPOR
           raise ArgumentError, "unsupported compression key #{compression}" unless compression.nil? || %w[gzip none].include?(compression)
 
           @uri = if endpoint == ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
-                   URI("#{endpoint}/v1/traces")
+                   URI(endpoint.end_with?('/') ? "#{endpoint}#{TRACES_PATH}" : "#{endpoint}/#{TRACES_PATH}")
```

Comment:
What do you think about using `File.join` here? I think that would eliminate the ternary and the string interpolation

```ruby
irb(main):001:0> File.join("http://localhost:4318", "v1/traces")
=> "http://localhost:4318/v1/traces"
irb(main):002:0> File.join("http://localhost:4318/", "v1/traces")
=> "http://localhost:4318/v1/traces"
```